src/frontend/src/components/DownloadButton.tsx

```
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Box, IconButton, Tooltip } from '@mui/material';
+import DownloadIcon from '@mui/icons-material/Download';
+import { useDownloadFile } from '../hooks/part-review.hooks';
+
+interface DownloadButtonProps {
+  fileId: string;
+  filename: string;
+  stopPropagation?: boolean;
+}
+
+const DownloadButton: React.FC<DownloadButtonProps> = ({ fileId, filename = 'download', stopPropagation = false }) => {
+  const { data: blobData } = useDownloadFile(fileId);
+
+  const handleDownload = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (stopPropagation) {
+      e.stopPropagation();
+      e.preventDefault();
+    }
+    if (!blobData) return;
```

We should throw an error here